h2o-py/h2o/frame.py

```diff
@@ -1280,34 +1284,63 @@ def drop(self, index, axis=1):
+
                 fr = H2OFrame._expr(expr=ExprNode("cols", self, -(index + 1)), cache=self._ex._cache)
                 fr._ex._cache.ncols -= 1
                 fr._ex._cache.names = self.names[:index] + self.names[index + 1:]
                 fr._ex._cache.types = {name: self.types[name] for name in fr._ex._cache.names}
                 return fr
-            if all(isinstance(item, int) for item in index):
-                for i in range(len(index)):
+
+            elif isinstance(index,list):
+                #If input is an int array indicating a column index, i.e., [3] or [1,2,3]:
+                if all(isinstance(item, int) for item in index):
+                    assert max(index) < self.ncol, "Column index selected to drop is not part of the frame: %r" % index
+                    assert min(index) >= 0, "Column index selected to drop is not positive: %r" % index
+                    for i in range(len(index)):
                         index[i] = index[i] + 1
-                index = [-x for x in index]
-                fr = H2OFrame._expr(expr=ExprNode("cols", self, index), cache=self._ex._cache)
-            elif all(isinstance(item, str) for item in index):
-                for i in range(len(index)):
-                    if is_type(index[i], str):
-                        index[i] = self.names.index(index[i]) + 1
-                index = [-x for x in index]
+                    index = [-x for x in index]
+                #If index is a string array, i.e., ["C1", "C2"]
+                elif all(isinstance(item, str) for item in index):
+                    #Check if index is an actual column(s) in the frame
+                    assert set(index).issubset(self.names) == True, "Column(s) selected to drop are not in original " \
+                                                            "frame: %r" % index
+                    for i in range(len(index)):
+                        if is_type(index[i], str):
+                            index[i] = self.names.index(index[i]) + 1
+                    index = [-x for x in index]
+
                 fr = H2OFrame._expr(expr=ExprNode("cols", self, index), cache=self._ex._cache)
+                fr._ex._cache.ncols -= len(index)
+                fr._ex._cache.names[:] = [i for i in self.names]
+                fr._ex._cache.types = {name: fr.types[name] for name in fr._ex._cache.names}
+
             else:
-                raise ValueError("Invalid column index types. Must either be a list of all int indexes or "
-                                 "a list of all column names (strings) for dropping columns.")
+                raise ValueError("Invalid column index types. Must either be a list of all int indexes, "
+                                 "a string list of all column names, a single int index, or"
+                                 "a single string for dropping columns.")
             return fr
         elif axis == 0:
```

Comment:
I have serious doubts about this functionality... Do people ever want to drop few rows? Maybe sometimes, but more often they'd want to drop a slice...